### PR TITLE
[FEMDEM] Correcting pressure extrapolation process

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
In this PR I'm correcting (re-implementing) the pressure extrapolation process in 2D

### CODE REVIEW

File: applications/FemToDemApplication/CMakeLists.txt [hunk 1]
```diff
@@ -28,6 +28,7 @@ set( KRATOS_FEM_TO_DEM_APPLICATION_SOURCES
 	${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/femdem3d_hexahedron_element.cpp
 	${CMAKE_CURRENT_SOURCE_DIR}/custom_processes/extend_pressure_condition_process.cpp
 	${CMAKE_CURRENT_SOURCE_DIR}/custom_processes/assign_pressure_id_process.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_processes/expand_wet_nodes_process.cpp
```
Comment (loumalouomega) [inline applications/FemToDemApplication/CMakeLists.txt:31]: Wet...

File: applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//
+
+#include "custom_processes/expand_wet_nodes_process.h"
+
+namespace Kratos {
+
+ExpandWetNodesProcess::ExpandWetNodesProcess(
+    ModelPart& r_model_part)
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.cpp]: rModelPart->Input is cammel case

File: applications/FemToDemApplication/CMakeLists.txt (hunk 1 above)
Comment (AlejandroCornejo) [reply applications/FemToDemApplication/CMakeLists.txt:31]: was Francisco Zárate idea haha (it means that is in contact with the pressurfe load)

File: applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+#if !defined(KRATOS_EXPAND_WET_NODES_PROCESS)
+#define KRATOS_EXPAND_WET_NODES_PROCESS
+
+#include "includes/model_part.h"
+#include "processes/process.h"
+#include "fem_to_dem_application_variables.h"
+#include "includes/define.h"
+#include "includes/kratos_flags.h"
+#include <list>
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.h]: // System includes
#include <list>

// External includes

// Project includes
#include "includes/model_part.h"
#include "processes/process.h"
#include "fem_to_dem_application_variables.h"
#include "includes/define.h"
#include "includes/kratos_flags.h"

File: applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.h [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    /// Pointer definition of ExpandWetNodesProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ExpandWetNodesProcess);
+
+    // Constructor
+    ExpandWetNodesProcess(ModelPart &r_model_part);
+
+    // Destructor
+    ~ExpandWetNodesProcess() override = default;
+
+    void operator()() { Execute(); }
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.h:53]: Add Doxygen doc

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp [hunk 5]
```diff
@@ -12,218 +12,270 @@
...
 #include "custom_processes/extend_pressure_condition_process.h"
+#include "processes/find_elements_neighbours_process.h"
 
 namespace Kratos {
 
 template <SizeType TDim>
 ExtendPressureConditionProcess<TDim>::ExtendPressureConditionProcess(
-    ModelPart &r_model_part)
-    : mr_model_part(r_model_part)
+    ModelPart& r_model_part)
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: CammelCase

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp [hunk 6]
```diff
@@ -12,218 +12,270 @@
...
+template <>
+void ExtendPressureConditionProcess<2>::Execute()
+{
+    // We search the neighbours for the generation of line loads
+    auto find_neigh = FindElementalNeighboursProcess(mrModelPart, 2, 5);
+    find_neigh.Execute();
+    auto& r_process_info = mrModelPart.GetProcessInfo();
+
+    // Remove previous line loads-> Only the 1st iteration
+    if (r_process_info[ITER] == 1) {
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: Huuuum, is this the non linear iteration, the step?, the NR uses NL_ITERATION_NUMBER

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.h [hunk 7]
```diff
@@ -31,7 +31,7 @@ class ExtendPressureConditionProcess : public Process
 {
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.h:40]: Add Doxygen

File: applications/FemToDemApplication/custom_python/fem_to_dem_python_application.cpp [hunk 8]
```diff
@@ -95,6 +92,7 @@ namespace Python
 	KRATOS_REGISTER_IN_PYTHON_VARIABLE(m,NODAL_FORCE_Z);
```
Comment (loumalouomega) [inline applications/FemToDemApplication/custom_python/fem_to_dem_python_application.cpp:92]: Very specific variable...

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Some comments

File: applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.cpp (hunk 2 above)
Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.cpp]: done

File: applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.h (hunk 3 above)
Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_processes/expand_wet_nodes_process.h]: done

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp (hunk 5 above)
Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: ok

File: applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp (hunk 6 above)
Comment (AlejandroCornejo) [reply applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp]: nope, it is an internal variable that ensures several itartions when creating the new pressure loads

**Comment by loumalouomega** [comment]
Please change ITER name of the variable for something more descriptive like.  INTERNAL_PRESSURE_ITERATION or something more suitable

**Review by loumalouomega — DISMISSED** [review]
OK

**Review by loumalouomega — APPROVED** [review]

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**